STUDIO/org.openl.rules.workspace/src/org/openl/rules/project/abstraction/RulesProject.java

```diff
@@ -594,4 +601,53 @@ public String getMainBusinessName() {
         }
         return folderPath.substring(folderPath.lastIndexOf('/') + 1);
     }
+    
+    private final Object tagsLock = new Object();
+    
+    public Map<String, String> getTags() {
+        synchronized (tagsLock) {
+            if (tags == null) {
+                tags = loadTags();
+            }
+            return tags;
+        }
+    }
```

Comment:
You should minimize the scope of the synchronization block by checking if whatever tags are initialized or not. Also it's possible to eliminate mutex object by using `this` in synchronized block.
```suggestion
public Map<String, String> getTags() {
    var localTags = tags;
    if (localTags == null) {
        synchronized (this) {
            localTags = tags;
            if (localTags == null) {
                localTags = loadTags();
                tags = localTags;
            }
        }
    }
    return localTags;
}
```